File: java/src/LivingThings/Animals/Kambing.java
import java.util.concurrent.ThreadLocalRandom;

// nextInt is normally exclusive of the top value,
// so add 1 to make it inclusive

public class Kambing extends FarmAnimal implements LivingThings{
   char t_rumput ='#';
   char tempat ='-';
   char produk ='M';

    final int randX =1;
    final int randY =1;

    //private int Full;
    private int posX;
    private int posY;
    private int countgoatMilk;

    //Renderable c;

    public Kambing (int posX ,int posY,Renderable c){
       
        super(posX,posY);
       // c.setElement(posX,posY)
        //Cell c = new Cell();
        c.setElement(posX,posY,'K');
        this.countgoatMilk =0;
        this.posX = posX;
        this.posY = posY;
        //this.Full = FULLMAX;


    }
    public void move (Renderable c ){
        int x = ThreadLocalRandom.current().nextInt(-randX, randY + 1);
        int y =ThreadLocalRandom.current().nextInt(-randX, randY);
        //System.out.println("x : " + x);
        //System.out.println("y : " +y);
        
        if (this.posX+x>=0 && this.posX+x<15&& this.posY+y>=0 && this.posY +y <15 && hidup() && c.getElement(this.posX+x,this.posY+y)== ' '){
            
            if (c.getTypeLand(this.posX+x,this.posY+y)==tempat){
                c.setElement(posX,posY,' ');
                this.posX+=x;
                this.posY +=y;
                c.setElement(this.posX,this.posY,'K');
                setFull(getFull()-1);
                
            } 
            
            //System.out.println(getFull());
            //c.setElement(this.posX,this.posY,'K');
        } else {
            if (!hidup()){
                c.setElement(this.posX,this.posY,' ');
            }
            
        }
    }
    public void talk() {
        System.out.println("MBEKKKK Mbekkk");
    }
    public void eat(Renderable c){
        if (!hidup()){
                //setStatus(false);
                c.setElement(this.posX,this.posY,' ');
        } else {
            if(c.getRumput(this.posX,this.posY)==t_rumput){
                this.countgoatMilk =MAX;
                setFull(FULLMAX);
                c.kosongRumput(this.posX,this.posY);
         } else move(c);
        
        }
       
    }
    public String getChar() {
        return "K";
    }

    public void getProduct() {
        this.countgoatMilk =0;
    }
    public Integer getCountProduct() {
        return countgoatMilk;
    }
    /*public Integer getFull() {
        return Full;
    }*/

    /*public void setFull(int Full){
        this.Full = Full;
    }*/
    public void print() {
        System.out.println("kambing " + countgoatMilk + " Full : "+ getFull()+"pos :  "+ posX +" "+ posY);
    }
}

File: src/kambing.cpp
#include <iostream>
#include <string>
#include "FarmAnimal.h"
#include "kambing.h"
#include "Milkproducing.h"
#include "Meatproducing.h"

using namespace std;
const int a = 10;
const int b = 4;
const int c =5;
const int d = 6;

kambing::kambing(int posX , int posY):FarmAnimal(posX, posY), Meatproducing(posX, posY), Milkproducing(posX, posY) {
    Full = FULLMAX;
    countgoatMilk=0;
    this->posX = FarmAnimal::getPosX();
    this->posY = FarmAnimal::getPosY();
}

kambing::~kambing() {
    this->posX = 0;
    this->posY = 0;
    Full =0;
    countgoatMilk=0;
    //cout<<"One of your goats has died"<<endl;
}

void kambing::move(Cell&_c) {
    int xa = FarmAnimal::getPosX();
    int x = rand()%(a+b-1);
    int y = rand()%(a+b-1);
    _c.setElement(posX,posY,' ');
    //cout<<" x , y "<<x<< " " <<y<<endl;
    xa+=x;
    this->posY +=y;
    while(xa>a) {
         x = rand()%(a+b-1);
        //cout<<" x "<<x<<endl;
        while (this->posY>b) {
            this->posY -=x;
        }
        this->posX -=x;
        if (xa<a) {
            xa = xa + a;
        }
    }
    FarmAnimal::setPosX(xa);
    _c.setElement(posX,posY,'A');
    cout<<"pos MOVE : "<< this->posX<<" " << this->posY<<endl;
    
    Full--;
}
void kambing::talk() {
    cout<<"Mbekkkk mbeek"<<endl;
}

void kambing::eat(Cell& _c) {
    if (_c.getRumput(this->posX,this->posY)){
        Full = FULLMAX;
        countgoatMilk++;
        if (countgoatMilk > MAX) {
            countgoatMilk = MAX;
        }
        _c.setElement(posX,posY,' ');
    }
}

string kambing::getChar() {
    return "K";
}

string kambing::getProduct() {
    return "";
}

//void setProduct(string goatMilk);
int kambing::getcountgoatMilk() const {
    return countgoatMilk;
}

int kambing::getFull() {
    return Full;
}

void kambing::setFull(int Full) {
    this->Full=Full;
}

void kambing::Print() {
    cout<<"GM ada " <<countgoatMilk<<endl;
    cout<<"Full : "<< Full<<endl;
    cout<<"pos : "<< this->posX << " " << this->posY<<endl;
}

/*void kambing::setPosX(int x) {
    posX = x;
}

void kambing::setPosY(int y) {
    posY= y;
}

int kambing ::getPosX() {
    return posX;
}

int kambing ::getPosY() {
    return posY;
}
*/
